skimage/morphology/extrema.py

```diff
@@ -207,152 +210,358 @@ def h_minima(image, h, selem=None):
+
+    Parameters
+    ----------
+    image : ndarray
+        The array to modify inplace.
+    value : number
+        The value to use. Should be compatible with `image`'s dtype.
+
+    Examples
+    --------
+    >>> image = np.zeros((4, 5), dtype=int)
+    >>> _set_edge_values_inplace(image, 1)
+    >>> image
+    array([[1, 1, 1, 1, 1],
+           [1, 0, 0, 0, 1],
+           [1, 0, 0, 0, 1],
+           [1, 1, 1, 1, 1]])
     """
-    img_vec = np.unique(image.flatten())
+    for axis in range(image.ndim):
+        sl = [slice(None)] * image.ndim
+        # Set edge in front
+        sl[axis] = 0
+        image[sl] = value
+        # Set edge to the end
+        sl[axis] = -1
+        image[sl] = value
+
+
+def _fast_pad(image, value):
+    """Pad an array on all axes with one constant value.
+
+    Parameters
+    ----------
+    image : ndarray
+        Image to pad.
+    value : number
+         The value to use. Should be compatible with `image`'s dtype.
 
-    if img_vec.size == 1:
-        return 0
+    Returns
+    -------
+    padded_image : ndarray
+        The new image.
 
-    min_diff = np.min(img_vec[1:] - img_vec[:-1])
-    return min_diff
+    Notes
+    -----
+    The output of this function is equivalent to::
 
+        np.pad(image, mode="constant", constant_values=value)
 
-def local_maxima(image, selem=None):
-    """Determine all local maxima of the image.
+    However this method needs to only allocate and copy once which can result
+    in significant speed gains if `image` is large.
 
-    The local maxima are defined as connected sets of pixels with equal
-    grey level strictly greater than the grey levels of all pixels in direct
-    neighborhood of the set.
+    Examples
+    --------
+    >>> _fast_pad(np.zeros((2, 3)), 4)
+    array([[4., 4., 4., 4., 4.],
+           [4., 0., 0., 0., 4.],
+           [4., 0., 0., 0., 4.],
+           [4., 4., 4., 4., 4.]])
+    """
+    # Allocate padded image
+    new_shape = np.array(image.shape) + 2
+    new_image = np.empty(new_shape, dtype=image.dtype)
+
+    # Copy old image into new space
+    original_slice = tuple(slice(1, -1) for _ in range(image.ndim))
+    new_image[original_slice] = image
+    # and set the edge values
+    _set_edge_values_inplace(new_image, value)
 
-    For integer typed images, this corresponds to the h-maxima with h=1.
-    For float typed images, h is determined as the smallest difference
-    between grey levels.
+    return new_image
+
+
+def _offset_to_raveled_neighbours(image_shape, selem):
+    """Compute offsets to a samples neighbors if the image would be raveled.
 
     Parameters
     ----------
-    image : ndarray
-        The input image for which the maxima are to be calculated.
+    image_shape : tuple
+        The image for which the offsets are computed.
     selem : ndarray, optional
         The neighborhood expressed as an n-D array of 1's and 0's.
         Default is the ball of radius 1 according to the maximum norm
```

Comment:
default has to be removed as well...

Reply:
the `selem` parameter in this signature `_offset_to_raveled_neighbours(image_shape, selem)` is not optional.

Reply:
Ah, sorry. :see_no_evil:  I thought this was a comment in local_maxima's docstring.

Reply:
no problem!